### PR TITLE
lib: cmsis_rtos_v1: Minor refactor of CMSIS implementation

### PR DESCRIPTION
**Pull request by pswarnak**
Add few missing NULL checks to avoid crash. Also, minor
refactor of signal code and disable osFeature_Wait to
signify osWait function not implemented.

Signed-off-by: Praful Swarnakar <praful.swarnakar@intel.com>

### CODE REVIEW

**Comment by pswarnak** [comment]
The shippable failure is due to checkpatch warning for the external file: cmsis_os.h
Kindly ignore it as the modification is only the change of default value from 1 to 0.
We are not changing the formatting to keep the original file similar to the zephyr ported file.

File: lib/cmsis_rtos_v1/cmsis_mailq.c [hunk 1]
```diff
@@ -167,6 +167,11 @@ osStatus osMailFree(osMailQId queue_id, void *mail)
 {
 	osMailQDef_t *queue_def = (osMailQDef_t *)queue_id;
 
+	/* Note: Below 2 error codes are not supported.
+	 * osErrorValue: mail block does not belong to the mail queue pool.
+	 * osErrorParameter: the value to the parameter queue_id is incorrect.
+	 */
+
```
Comment (rgundi) [inline lib/cmsis_rtos_v1/cmsis_mailq.c]: Remove these comments. They will be put in the doc section separately.

File: lib/cmsis_rtos_v1/cmsis_thread.c [hunk 2]
```diff
@@ -49,7 +49,7 @@ osThreadId osThreadCreate(const osThreadDef_t *thread_def, void *arg)
 		 thread_def->stacksize <= CONFIG_CMSIS_THREAD_MAX_STACK_SIZE,
 		 "invalid stack size\n");
 
-	if (thread_def->instances == 0) {
+	if ((!thread_def) || (thread_def->instances == 0)) {
```
Comment (rgundi) [inline lib/cmsis_rtos_v1/cmsis_thread.c]: Use thread_def == NULL to maintain uniformity

File: lib/cmsis_rtos_v1/cmsis_thread.c [hunk 3]
```diff
@@ -101,7 +101,7 @@ osPriority osThreadGetPriority(osThreadId thread_id)
 	k_tid_t thread = (k_tid_t)thread_id;
 	u32_t priority;
 
-	if (_is_in_isr()) {
+	if ((!thread_id) || (_is_in_isr())) {
```
Comment (rgundi) [inline lib/cmsis_rtos_v1/cmsis_thread.c]: Use thread_id == NULL to preserve uniformity

File: lib/cmsis_rtos_v1/cmsis_signal.c [hunk 4]
```diff
@@ -79,7 +81,7 @@ osEvent osSignalWait(int32_t signals, uint32_t millisec)
 	}
 
 	/* Check if signals is within the permitted range */
-	if (signals >= (1 << (osFeature_Signals + 1))) {
+	if ((!signals) || (signals >= (1 << (osFeature_Signals + 1)))) {
```
Comment (rgundi) [inline lib/cmsis_rtos_v1/cmsis_signal.c]: signals = 0 is a valid case. Undo this change.

**Review by rgundi — CHANGES_REQUESTED** [review]

File: lib/cmsis_rtos_v1/cmsis_mailq.c (hunk 1 above)
Comment (pswarnak) [reply lib/cmsis_rtos_v1/cmsis_mailq.c]: I saw similar comments mentioned in Mutex, Semaphore and Mempool.
These lines were taken from lib/cmsis_rtos_v1/cmsis_mempool.c
Shall i remove it even in those places ?

**Review by rgundi — APPROVED** [review]
LGTM

File: lib/cmsis_rtos_v1/cmsis_mailq.c (hunk 1 above)
Comment (pswarnak) [reply lib/cmsis_rtos_v1/cmsis_mailq.c]: Got it. Addressed the comment.

File: lib/cmsis_rtos_v1/cmsis_thread.c (hunk 2 above)
Comment (pswarnak) [reply lib/cmsis_rtos_v1/cmsis_thread.c]: Done

File: lib/cmsis_rtos_v1/cmsis_thread.c (hunk 3 above)
Comment (pswarnak) [reply lib/cmsis_rtos_v1/cmsis_thread.c]: Done

File: lib/cmsis_rtos_v1/cmsis_signal.c (hunk 4 above)
Comment (pswarnak) [reply lib/cmsis_rtos_v1/cmsis_signal.c]: Sorry, my bad, missed reading that part. Code changed accordingly.

**Comment by pswarnak** [comment]
@rgundi: Thanks for the quick review and approval.

**Comment by pswarnak** [comment]
@nashif: Kindly review/merge. Shippable failure is due to checkpatch warning for the external file: cmsis_os.h

**Review by nashif — APPROVED** [review]
